src/controllers/emotes/EmoteProvider.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "messages/Emote.hpp"
+#include "util/Expected.hpp"
+
+#include <pajlada/signals/signal.hpp>
+#include <QString>
+
+#include <memory>
+
+namespace chatterino {
+
+class EmoteChannel;
+
+struct EmoteLoadResult {
+    EmoteMapPtr emotes;
+    bool isPreemptiveCached = false;
+};
+
+class EmoteProvider
```

Comment:
What do you foresee the derived classes of EmoteProvider to be?

Reply:
Yes.